### PR TITLE
CXX-1030 Require C driver 1.5.0 and build against C driver master bra

### PR DESCRIPTION
**Pull request by acmorrow**
…nch to obtain it

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -59,7 +59,14 @@ install:
     - sudo apt-get install cmake
 
     # Install Mongo C Driver
-    - pushd mongo-c-driver; ./autogen.sh --enable-tests=no --enable-examples=no --with-libbson=bundled; make; sudo make install; popd;
+    - pushd mongo-c-driver
+
+    # TODO: Update this to our real minimum for the C++11 driver 3.1 release, once known.
+    - git checkout master
+
+    - ./autogen.sh --enable-tests=no --enable-examples=no --with-libbson=bundled; make; sudo make install
```
Comment (xdg) [inline .travis.yml]: I think this should match the autogen line in .mci.yml, right?  At least, it should have the `--disable-automatic-init-and-cleanup` flag.

Comment (acmorrow) [inline .travis.yml]: Eh, maybe. We only *really* need that when doing the ASAN builds, and we don't do those on travis. Maybe it is better to ensure that the driver tests pass when the C driver is built both ways? In the real world, the C++11 driver will encounter both C driver modes, at least for now.

File: .mci.yml [hunk 2]
```diff
@@ -77,9 +77,10 @@ functions:
                 set -o verbose
                 git clone https://github.com/mongodb/mongo-c-driver.git
                 cd mongo-c-driver
-                git checkout 1.3.4
+                # TODO: Update this to our real minimum for the C++11 driver 3.1 release, once known.
+                git checkout master
                 rm -rf /data/tmp/c-driver-install
-                ./autogen.sh --prefix="/data/tmp/c-driver-install" --enable-tests=no --enable-examples=no --with-libbson=bundled --disable-extra-align --enable-debug --enable-optimizations --disable-shm-counters --disable-static --disable-dependency-tracking --with-pic --disable-automatic-init-and-cleanup ${cdriver_configure_flags}
+                ./autogen.sh --prefix="/data/tmp/c-driver-install" --enable-tests=no --enable-examples=no --with-libbson=bundled --disable-decimal128 --disable-extra-align --enable-debug --enable-optimizations --disable-shm-counters --disable-static --disable-dependency-tracking --with-pic --disable-automatic-init-and-cleanup ${cdriver_configure_flags}
```
Comment (acmorrow) [inline .mci.yml]: Note that we are now disabling decimal128 until CDRIVER fixes their headers.

File: .travis.yml [hunk 3]
```diff
@@ -59,7 +58,14 @@ install:
     - sudo apt-get install cmake
 
     # Install Mongo C Driver
-    - pushd mongo-c-driver; ./autogen.sh --enable-tests=no --enable-examples=no --with-libbson=bundled; make; sudo make install; popd;
+    - pushd mongo-c-driver
+
+    # TODO: Update this to our real minimum for the C++11 driver 3.1 release, once known.
+    - git checkout master
+
+    - ./autogen.sh --enable-tests=no --enable-examples=no --with-libbson=bundled --disable-decimal128; make; sudo make install
```
Comment (acmorrow) [inline .travis.yml]: Here too for disabling decimal.

**Comment by acmorrow** [comment]
The latest version of this has passed travis and appveyor. @xdg PTAL. Thanks.

**Comment by acmorrow** [comment]
@xdg - It didn't seem to need to. I think perhaps MSVC doesn't offer _Decimal128

File: .mci.yml [hunk 4]
```diff
@@ -77,9 +77,10 @@ functions:
                 set -o verbose
                 git clone https://github.com/mongodb/mongo-c-driver.git
                 cd mongo-c-driver
-                git checkout 1.3.4
+                # TODO: Update this to our real minimum for the C++11 driver 3.1 release, once known.
+                git checkout master
                 rm -rf /data/tmp/c-driver-install
-                ./autogen.sh --prefix="/data/tmp/c-driver-install" --enable-tests=no --enable-examples=no --with-libbson=bundled --disable-extra-align --enable-debug --enable-optimizations --disable-shm-counters --disable-static --disable-dependency-tracking --with-pic --disable-automatic-init-and-cleanup ${cdriver_configure_flags}
+                ./autogen.sh --prefix="/data/tmp/c-driver-install" --enable-tests=no --enable-examples=no --with-libbson=bundled --disable-decimal-bid --disable-extra-align --enable-debug --enable-optimizations --disable-shm-counters --disable-static --disable-dependency-tracking --with-pic --disable-automatic-init-and-cleanup ${cdriver_configure_flags}
```
Comment (xdg) [inline .mci.yml]: This is a test comment to test the new PR review UI.  I guess I click "start a review" now.

File: .travis.yml [hunk 5]
```diff
@@ -59,7 +58,14 @@ install:
     - sudo apt-get install cmake
 
     # Install Mongo C Driver
-    - pushd mongo-c-driver; ./autogen.sh --enable-tests=no --enable-examples=no --with-libbson=bundled; make; sudo make install; popd;
+    - pushd mongo-c-driver
+
+    # TODO: Update this to our real minimum for the C++11 driver 3.1 release, once known.
+    - git checkout master
```
Comment (xdg) [inline .travis.yml]: This is another test comment.

File: .mci.yml (hunk 4 above)
Comment (acmorrow) [inline .mci.yml]: ack

File: .travis.yml (hunk 5 above)
Comment (acmorrow) [inline .travis.yml]: ack

**Comment by xdg** [comment]
LGTM as a temporary workaround.  I have mixed feelings about allowing destructors at all, but I guess we can reconsider it after [CXX-1032](https://jira.mongodb.org/browse/CXX-1032).

**Pull request merged**